### PR TITLE
[pytest] Port advanced reboot based test cases (sad path)

### PR DESCRIPTION
**Pull request by tahmed-dev**
### Description of PR
This patch ports remaining sad path advanced reboot test cases
to pytest infra.

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### CODE REVIEW

File: tests/common/fixtures/advanced_reboot.py [hunk 1]
```diff
@@ -408,8 +408,8 @@ def __runPtfRunner(self, rebootOper=None):
         '''
         logger.info("Running PTF runner on PTF host: {0}".format(self.ptfhost))
 
-        prebootOper = rebootOper if rebootOper is not None and 'routing' in rebootOper else None
-        inbootOper = rebootOper if rebootOper is not None and 'routing' not in rebootOper else None
+        prebootOper = rebootOper if rebootOper is not None and 'routing' not in rebootOper else None
+        inbootOper = rebootOper if rebootOper is not None and 'routing' in rebootOper else None
```
Comment (yxieca) [inline tests/common/fixtures/advanced_reboot.py:412]: Please add a comments here to explain the calculation of preboot and inboot.

File: tests/platform/test_advanced_reboot.py [hunk 2]
```diff
@@ -15,3 +15,152 @@ def test_warm_reboot(request, get_advanced_reboot):
...
+    advancedReboot = get_advanced_reboot(rebootType='warm-reboot')
+    prebootList = [
+        'neigh_bgp_down',
+        'dut_bgp_down',
+        'dut_lag_down',
+        'neigh_lag_down',
+        'dut_lag_member_down:1:1',
+        'neigh_lag_member_down:1:1',
+        'vlan_port_down',
+    ]
```
Comment (yxieca) [inline tests/platform/test_advanced_reboot.py:36]: Add comments for each sad option?

File: tests/platform/test_advanced_reboot.py [hunk 3]
```diff
@@ -15,3 +15,152 @@ def test_warm_reboot(request, get_advanced_reboot):
...
+        'neigh_bgp_down:2',
+        'dut_bgp_down:3',
+        'dut_lag_down:2',
+        'neigh_lag_down:3',
+        'dut_lag_member_down:3:1',
+        'neigh_lag_member_down:2:1',
+        'vlan_port_down:4',
+    ] + ([
+        'dut_lag_member_down:2:{0}'.format(lagMemberCnt),
+        'neigh_lag_member_down:3:{0}'.format(lagMemberCnt),
```
Comment (yxieca) [inline tests/platform/test_advanced_reboot.py:63]: Comments for each sad opton?

File: tests/platform/test_advanced_reboot.py [hunk 4]
```diff
@@ -15,3 +15,152 @@ def test_warm_reboot(request, get_advanced_reboot):
...
+def test_warm_reboot_multi_sad_inboot(request, get_advanced_reboot):
+    '''
+    Warm reboot with multi sad path (during boot)
+
+    inboot list format: 'inboot_oper:route_cnt'
+    '''
+    advancedReboot = get_advanced_reboot(rebootType='warm-reboot')
+    inbootList = [
+        'routing_del:50',
+        'routing_add:50',
```
Comment (yxieca) [inline tests/platform/test_advanced_reboot.py:81]: comments please?

File: tests/platform/test_advanced_reboot.py [hunk 5]
```diff
@@ -15,3 +15,152 @@ def test_warm_reboot(request, get_advanced_reboot):
...
+    '''
+    Warm reboot with sad (bgp)
+
+    preboot_list format is 'preboot oper type:number of VMS down:number of lag members down'.
+    For non lag member cases, this parameter will be skipped
+    '''
+    advancedReboot = get_advanced_reboot(rebootType='warm-reboot')
+    prebootList = [
+        'neigh_bgp_down:2',
+        'dut_bgp_down:3',
```
Comment (yxieca) [inline tests/platform/test_advanced_reboot.py:100]: comments?

File: tests/platform/test_advanced_reboot.py [hunk 6]
```diff
@@ -15,3 +15,152 @@ def test_warm_reboot(request, get_advanced_reboot):
...
+    For non lag member cases, this parameter will be skipped
+    '''
+    advancedReboot = get_advanced_reboot(rebootType='warm-reboot')
+    lagMemberCnt = advancedReboot.getlagMemberCnt()
+    prebootList = [
+        'dut_lag_member_down:3:1',
+        'neigh_lag_member_down:2:1',
+    ] + ([
+        'dut_lag_member_down:2:{0}'.format(lagMemberCnt),
+        'neigh_lag_member_down:3:{0}'.format(lagMemberCnt),
```
Comment (yxieca) [inline tests/platform/test_advanced_reboot.py:123]: comments?

File: tests/platform/test_advanced_reboot.py [hunk 7]
```diff
@@ -15,3 +15,152 @@ def test_warm_reboot(request, get_advanced_reboot):
...
+    '''
+    Warm reboot with sad path (lag)
+
+    preboot_list format is 'preboot oper type:number of VMS down:number of lag members down'.
+    For non lag member cases, this parameter will be skipped
+    '''
+    advancedReboot = get_advanced_reboot(rebootType='warm-reboot')
+    prebootList = [
+        'dut_lag_down:2',
+        'neigh_lag_down:3',
```
Comment (yxieca) [inline tests/platform/test_advanced_reboot.py:142]: comments?

File: tests/platform/test_advanced_reboot.py [hunk 8]
```diff
@@ -15,3 +15,152 @@ def test_warm_reboot(request, get_advanced_reboot):
...
+def test_warm_reboot_sad_vlan_port(request, get_advanced_reboot):
+    '''
+    Warm reboot with sad path (vlan port)
+
+    preboot_list format is 'preboot oper type:number of VMS down:number of lag members down'.
+    For non lag member cases, this parameter will be skipped
+    '''
+    advancedReboot = get_advanced_reboot(rebootType='warm-reboot')
+    prebootList = [
+        'vlan_port_down:4',
```
Comment (yxieca) [inline tests/platform/test_advanced_reboot.py:160]: comment?

File: tests/common/fixtures/advanced_reboot.py (hunk 1 above)
Comment (tahmed-dev) [reply tests/common/fixtures/advanced_reboot.py:416]: Done

File: tests/platform/test_advanced_reboot.py (hunk 3 above)
Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py:78]: ACK

File: tests/platform/test_advanced_reboot.py (hunk 4 above)
Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py]: Done

Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py]: Ack

File: tests/platform/test_advanced_reboot.py (hunk 3 above)
Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py:78]: Done

File: tests/platform/test_advanced_reboot.py (hunk 5 above)
Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py]: done

File: tests/platform/test_advanced_reboot.py (hunk 6 above)
Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py:152]: done

File: tests/platform/test_advanced_reboot.py (hunk 7 above)
Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py]: Done

File: tests/platform/test_advanced_reboot.py (hunk 8 above)
Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py]: done

File: tests/platform/test_advanced_reboot.py (hunk 2 above)
Comment (tahmed-dev) [reply tests/platform/test_advanced_reboot.py:45]: done

File: tests/common/fixtures/advanced_reboot.py (hunk 1 above)
Comment (tahmed-dev) [reply tests/common/fixtures/advanced_reboot.py:416]: OK

**Pull request merged**